test/telemetry.spec.ts

```diff
@@ -148,3 +208,150 @@ describe('ConsoleTelemetryExporter', () => {
     expectNoCalls(mockConsole, 'warn');
   });
 });
+
+describe('Direct telemetry initialization', () => {
+  const originalConsole = globalThis.console;
+  let mockConsole: Console;
+
+  beforeEach(() => {
+    mockConsole = new Console();
+    (globalThis as any).console = mockConsole;
+    Telemetry.reset();
+  });
+
+  afterEach(() => {
+    (globalThis as any).console = originalConsole;
+  });
+
+  it('uses the default telemetry provider and console exporter by default', () => {
+    const provider = Telemetry.getInstance();
+    expectNoCalls(mockConsole, 'debug');
+
+    const span = provider.startSpan('test span');
+    provider.count('test count', 1);
+    span.end();
+
+    expectCall(mockConsole, 'debug', 2);
+  });
+
+  it('uses the default telemetry provider given a custom telemetry exporter', () => {
+    const mockExporter = new MockTelemetryExporter();
+    const provider = Telemetry.getInstance(undefined, mockExporter);
+    expectNoCalls(mockExporter, 'sendSpan');
+    expectNoCalls(mockExporter, 'sendCount');
+
+    const span = provider.startSpan('test span');
+    provider.count('test count', 1);
+    span.end();
+
+    expectCall(mockExporter, 'sendSpan', 1);
+    expectCall(mockExporter, 'sendCount', 1);
+  });
+
+  it('ignores a given custom telemetry exporter given a custom telemetry provider', () => {
+    const mockProvider = new MockTelemetryProvider();
+    const mockExporter = new MockTelemetryExporter();
+    const provider = Telemetry.getInstance(mockProvider, mockExporter);
+    // endSpan is defined on MockTelemetryProvider to track span.end() calls
+    expectNoCalls(mockProvider, 'endSpan');
+    expectNoCalls(mockProvider, 'count');
+
+    const span = provider.startSpan('test span');
+    provider.count('test count', 1);
+    span.end();
+
+    expectCall(mockProvider, 'endSpan', 1);
+    expectCall(mockProvider, 'count', 1);
+
+    // Exporter should be unused if a custom provider is given
+    expectNoCalls(mockExporter, 'sendSpan');
+    expectNoCalls(mockExporter, 'sendCount');
+  });
+});
+
+describe('Telemetry initialization from window', () => {
+  const originalConsole = globalThis.console;
+  let mockConsole: Console;
+  const win: any = (window as any);
+  let initializeFromWindow: () => void;
+
+  beforeEach(async () => {
+    mockConsole = new Console();
+    (globalThis as any).console = mockConsole;
+
+    Telemetry.reset();
+
+    // Clear any rules that may have been configured in other tests
+    Object.keys(win)
+      .filter((key) => key.startsWith('_dlo_rules'))
+      .forEach((key) => {
+        win[key] = [];
+      });
+
+    // We dynamically import the initialization function so we can remove rules
+    // beforehand. If we import the initialization function while rules from other
+    // tests are still on the global window, test output noise is generated due to
+    // unexpected data layer and rule state
+    initializeFromWindow = (await import('../src/embed/init')).default;
```

Comment:
This is unfortunate complexity to absorb in test setup here. We have other tests that set `window._dlo_rules*` values and don't clean up those values on the global window. Rather than updating all those tests (which is probably still a good idea), I took a defensive stance in these tests to limit overall PR surface.